swan-lake/learn-the-language/distinctive-language-features/advanced-general-purpose-language-features.md

```diff
@@ -666,6 +666,89 @@ int cp = ch.toCodePointInt();
+
+Regular expression templates support interpolation, which allows you to dynamically insert values into a regular expression pattern at specific sub-syntax contexts defined by the regular expression grammar.
+
+For example:
+```ballerina
+function createPattern(string name) {
+    string:RegExp pattern = re `[a-z]{3}|${name}`;
+}
+```
+In this example, the regular expression pattern includes the `name` parameter interpolated into the pattern. This allows for dynamic patterns that can change based on the input.
+
+### Unicode property escape
+
+Unicode property escapes allow matching characters based on their Unicode properties.  For instance, Unicode property escapes can be used to match emojis, punctuations, letters from specific languages or scripts, etc. 
+
+```ballerina
+// '\p' will match the property value.
+string:RegExp lowerCaseLetter = re `\p{Ll}`;
+
+// '\P' will match the negation of the property value.
+string:RegExp nonDigitChar = re `\P{N}`;
+
+// adding `gc=` inside the property is optional.
+string:RegExp punctuation = re `\p{gc=P}`;
+
+// adding `sc=` will indicate specifing an Unicode languages.
+string:RegExp latin = re `\p{sc=Latin}`;
+```
+
+### Non-capturing groups
+
+Ballerina regex supports non-capturing groups and flags to control the behavior of regular expression patterns. These allow you to group a set of regular expression constructs without capturing the matched text.
+
+```ballerina
+string:RegExp nonCapturingGrpPattern = re `(?im-s:hello.+world)`;
+```
+
+In this example, a non-capturing group pattern have multiple flags inside the parentheses. The `i` flag makes the pattern case-insensitive. The `m` flag makes `^` and `$` match the beginning and end of each line. The `s` flag makes the `.` metacharacters match any character including line terminators. The pattern will match strings like `hello\nworld`, `hello world`, and `HELLO \n WORld`.
+
+### Using regexp values
+
+The ``RegExp`` values can be used with the [RegExp](https://lib.ballerina.io/ballerina/lang.regexp) methods (``split``, ``findAll``, ``findAllGroups``, etc.) and with the [string](https://lib.ballerina.io/ballerina/lang.string) methods [includesMatch](https://lib.ballerina.io/ballerina/lang.string/latest#includesMatch) and [matches](https://lib.ballerina.io/ballerina/lang.string/latest#matches).
+
+Two constructs are utilized to provide the output of these lang library functions:
+- ``Span``: An object type that functions as a container for substrings.
+- ``Group``: A tuple comprising ``Span`` objects where the first member of the tuple always represents the entire match.
+
+```ballerina
+string[] names = re `,`.split("Bob,Frank,Will,Jack"); // ["Bob", "Frank", "Will", "Jack"]
+
+int patternCount = re `[bB].tt[a-z]*`.findAll("Butter was bought by Betty but the butter was bitter.").length(); // 4
+
+string result = re `0+`.replaceAll("10010011", "*"); // 1*1*11
```

Comment:
Shall we add a sample that we get a Groups result and how we get Span out of it and prints start,end and substring